### DISCUSSION TITLE
OpenAI net::ERR_CONTENT_DECODING_FAILED

### DISCUSSION THREAD

**Discussion by LimLLL** [open-webui/open-webui#2348] [General]
# Bug Report

## Description
When using the third-party openai, the result in the f12 console network returns net::ERR_CONTENT_DECODING_FAILED. But in docker logs, it returns 200 ok. The behavior was ok when using Python or other ai programs.

## Environment

- **Open WebUI Version:** v0.1.124
- **Ollama (if applicable):** 0.1.37

- **Operating System:** Ubuntu 20.04
- **Browser (if applicable):** Chrome 124.0.6367.92

## Reproduction Details

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [x] I have included the browser console logs.
- [x] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
[Include relevant Docker container logs, if applicable]

**Screenshots (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]
<img width="756" alt="image" src="https://github.com/open-webui/open-webui/assets/59259075/317284a4-33ab-4446-82b3-bfaec1de79b5">
![CleanShot 2024-05-17 at 12 36 23@2x](https://github.com/open-webui/open-webui/assets/59259075/04278b09-3601-4563-a5f4-f9319c0c71b7)

![CleanShot 2024-05-17 at 11 25 02@2x](https://github.com/open-webui/open-webui/assets/59259075/375659a0-6b57-43ab-9a1d-4f95b73fbc23)

## Installation Method
both ollama and openweb UI are installed through the docker

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by sp391607254** [comment]
When using the third-party openai, I'm also having the same issue, using the nginx reverse proxy, check out  open-webui  logs   return 200 。What should I do to deal with this

**Comment by xindoo** [comment]
I encountered a similar issue when calling the Qwen series models. One phenomenon I observed is that all /chat/completions requests included the header `Accept-Encoding: gzip, deflate, br, zstd`.

The response headers from OpenAI do not return `Content-Encoding: gzip`, but the responses from the Qwen series models do return `Content-Encoding: gzip`, which then results in a **net::ERR_CONTENT_DECODING_FAILED** error.

I'm not sure whether this issue needs to be fixed by open-webui or the Qwen model API.

**Reply by xindoo** [reply]
I have consulted with Alibaba, and they have informed me that this issue appears to be related to open-webui. When making a request to the chat/completions interface, an additional header of `Accept-Encoding: gzip, deflate, br, zstd` is being included, which causes the interface to return results that are compressed using gzip. It is possible that open-webui is not handling the gzip data correctly. After I reconfigured the Accept-Encoding header through a nginx reverse proxy, this problem was solved.

**Reply by GuanJdoJ** [reply]
请问你的nginx是怎么配置Accept-Encoding的

**Reply by xindoo** [reply]
> 请问你的nginx是怎么配置Accept-Encoding的

你更新下open-webui最新版，他们修复了这个问题

**Reply by GuanJdoJ** [reply]
我其实没有用open-webui，只是遇到一个类似的问题，搜索到了这个discussion，想类比看看能否解决

**Comment by sp391607254** [comment]
I've solved this problem now, and by using one-api to proxy the services of various openai APIs to the local area and then give them to open-webui to use, there is no problem

**Reply by LimLLL** [reply]
Hi, thanks for sharing. Is there another way to fix this? Since most third-party OpenAI API service providers already use One API, it's kind of redundant to deploy One API again locally.
